src/metadata.cpp

```diff
@@ -2434,7 +2434,9 @@ void Metadata::InternalData::update_indexes(int protocol_version, const VersionN
       table->clear_indexes();
```

Comment:
The main problem is here. If the table is not found the 'table_name' must be reset!
So, the line 
`if (!table) continue;`
must be changed into:
`  if (!table) {
     table_name.clear();
     continue;
   }`